### ISSUE TITLE
Training learning curve 

### ISSUE THREAD

**Issue by Pacomito** [KerenLab/CellSighter#5]
Hello,

Thank you for this great and promising tool.

While in some case the training seems to work well and results look great, sometimes the quality of predictions seem to be very impacted by which iteration you take the weights from.

In this simple example, I tried to classify cells into 2 classes based on 1 marker and DAPI.

I am surprised at how the "learning curve" evolves. I ran 100 iterations ("epoch_max": 101, "lr": 0.001) on 6 trained images (ROI-01 ... ROI-06). I calculate the F1 score for evaluations from 10 to 100 on both the training images and the validation images:

**F1 score of training images**
![image](https://github.com/KerenLab/CellSighter/assets/36704054/78028eca-2528-43a5-8fdb-08c0f084d5c5)

**F1 score of validation images**
![image](https://github.com/KerenLab/CellSighter/assets/36704054/40d5fdef-db1d-4f8f-9646-c8689678d6b0)

It seems to follow the error rate more or less, for instance a good quality is attained at epoch 30 and the error is only of 0.05 but then drops dramatically at epoch 50 and the error is 0.27 ( see below).

So my questions are:
-  Is this expected ? 
- What can I do to systematically find the optimal training iteration ?
- Does this mean I have to add more epochs or change the learning curve ? 

Any tips or insigth would be greatly appreciated.

Thank you,
Pacôme

_Training:_
_epoch 0 | iterate 0 / 7 | 0.7371284365653992
Epoch 0 done!
epoch 1 | iterate 0 / 7 | 0.9678863286972046
Epoch 1 done!
...
epoch 10 | iterate 0 / 7 | 0.4672516882419586
Epoch 10 done!
Eval 0 / 13
...
epoch 20 | iterate 0 / 7 | 0.1741466075181961
Epoch 20 done!
Eval 0 / 13
...
**epoch 30** | iterate 0 / 7 | **0.051422443240880966**
Epoch 30 done!
Eval 0 / 13
...
epoch 40 | iterate 0 / 7 | 0.11234039068222046
Epoch 40 done!
Eval 0 / 13
...
**epoch 50** | iterate 0 / 7 | **0.27082550525665283**
Epoch 50 done!
Eval 0 / 13
...
epoch 60 | iterate 0 / 7 | 0.08186761289834976
Epoch 60 done!
Eval 0 / 13
...
**epoch 70** | iterate 0 / 7 | **0.110440194606781**
Epoch 70 done!
Eval 0 / 13
...
epoch 80 | iterate 0 / 7 | 0.017165539786219597
Epoch 80 done!
Eval 0 / 13
...
epoch 90 | iterate 0 / 7 | 0.024419160559773445
Epoch 90 done!
Eval 0 / 13
...
**epoch 100** | iterate 0 / 7 | **0.025695184245705605**
Epoch 100 done!
Eval 0 / 13_

**Comment by yaelAmitay** [comment]
Hi,
It looks like your loss is fluctuating. It could be that you need to the lower learning rate or maybe change the lr decay to decrease faster.
If you could share the loss plot here it will help.

**Comment by Pacomito** [comment]
Hi, how would you plot that ?

Thank you

**Comment by yaelAmitay** [comment]
You can see it in the Tensorboard

**Comment by Pacomito** [comment]
Thank you for the suggestion,

Here is the Tensorboard for 500 iterations:

![image](https://github.com/KerenLab/CellSighter/assets/36704054/ac0d7ce2-5074-43da-ad94-bba09e7f69e7)

**Comment by yaelAmitay** [comment]
Do you mean 500 epochs?

In general it looks like the loss is decreasing, in train.py you have:
`    scheduler = torch.optim.lr_scheduler.ExponentialLR(optimizer, gamma=0.85)
`
It's currently not being used but you can try to add it to the training loop and it should help.
By applying once in a few epochs (or every epoch):
`
scheduler.step()
`

You can try training for a longer period and see if the loss and f1 score stabilize.

There are many issues that can cause this behaviour of the data it's difficult to say what cause it exactly.
